File: exercicios/data_rpg.c
#include <stdio.h>

enum CLASSE {GUERREIRO, ASSASSINO, SACERDOTE};
enum ARMA {ESPADA, ADAGAS, CAJADO};

struct Arma {
    enum ARMA tipo;
    
    union {
        struct {char nome_espada[50];};
        struct {char nome_adaga[50];};
        struct {char nome_cajado[50];};
    };
};

struct Classes {
    enum CLASSE classe;
    char nome[50];
    int altura;
    
    union {
        struct {struct Arma espada;}; //guerreiro
        struct {struct Arma adaga;}; //assassino
        struct {struct Arma cajado;}; //sacerdote
    };
};

int main() {
    struct Classes guerreiro = {GUERREIRO, "Ricardo", 190, {ESPADA, {"espada do aprendiz"}}};
    return 0;
}

File: exercicios/data_rpg.hs
-- Guerreiros: possuem nome, altura (em cm) e usam espada
-- Assassinos: possuem nome, altura (em cm) e usam adagas
-- Sacerdotes: possuem nome, altura (em cm) e usam cajado

data Arma = Espada
            | Adagas
            | Cajado
    deriving (Show)

data Classes a = Guerreiro String Int a
                | Assassino Classes a
                | Sacerdote Classes a
    deriving (Show)


guerreiro = Guerreiro "Ricardo" 190 Espada

main = print guerreiro
